app/routes/questionnaire.py

```diff
@@ -306,37 +301,12 @@ def _generate_wtf_form(block_schema, schema, current_location):
 
 
 def submit_answers(schema, questionnaire_store, full_routing_path):
-    metadata = questionnaire_store.metadata
-
-    message = json.dumps(
-        convert_answers(schema, questionnaire_store, full_routing_path), for_json=True
-    )
-
-    encrypted_message = encrypt(
-        message, current_app.eq["key_store"], KEY_PURPOSE_SUBMISSION
+    submission_handler = SubmissionHandler(
+        schema, questionnaire_store, full_routing_path
     )
-    sent = current_app.eq["submitter"].send_message(
-        encrypted_message,
-        questionnaire_id=metadata.get("questionnaire_id"),
-        case_id=metadata.get("case_id"),
-        tx_id=metadata.get("tx_id"),
-    )
-
-    if not sent:
-        raise SubmissionFailedException()
-
-    submitted_time = datetime.utcnow()
-    _store_submitted_time_in_session(submitted_time)
+    submission_handler.submit_message()
     get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
```

Comment:
I think this should either be done in the `submit_message` method, or another method created on the handler e.g. `delete_questionnaire_state`.

Reply:
If the above is done, I think this `submit_answers` should be scrapped as it doesn't provide much over making an instance of `SubmissionHandler` in the routes. This would be if the message was handled by submission as per https://github.com/ONSdigital/eq-questionnaire-runner/pull/151#discussion_r442030960

Reply:
I kept the delete state out of the handler as I considered them to be distinct things

Reply:
Not sure about duplication of the handler in hub/block submissions